File: ares-fabric/src/main/java/dev/tigr/ares/fabric/impl/modules/misc/MsgOnToggle.java
package dev.tigr.ares.fabric.impl.modules.misc;

import dev.tigr.ares.core.event.client.ToggleEvent;
import dev.tigr.ares.core.feature.module.Category;
import dev.tigr.ares.core.feature.module.Module;
import dev.tigr.ares.core.util.render.TextColor;
import dev.tigr.simpleevents.listener.EventHandler;
import dev.tigr.simpleevents.listener.EventListener;
import dev.tigr.simpleevents.listener.Priority;

/**
 * @author Tigermouthbear
 */
@Module.Info(name = "MsgOnToggle", description = "Sends a chat message when a module is toggled", category = Category.MISC)
public class MsgOnToggle extends Module {
    public static MsgOnToggle INSTANCE;

    public MsgOnToggle() {
        INSTANCE = this;
    }

    @EventHandler
    public EventListener<ToggleEvent> toggleEvent = new EventListener<>(event -> {
        if(event.getModule().getName().equalsIgnoreCase("clickgui")) return;
        String text = event.isEnabled() ? TextColor.GREEN + "Enabled " : TextColor.RED + "Disabled ";
        UTILS.printMessage(text + TextColor.BLUE + event.getModule().getName());
    });
}

File: src/main/kotlin/org/aresclient/ares/impl/instrument/module/modules/misc/MsgOnToggle.kt
package org.aresclient.ares.impl.instrument.module.modules.misc

import dev.tigr.simpleevents.listener.EventHandler
import dev.tigr.simpleevents.listener.EventListener
import org.aresclient.ares.api.events.ToggleEvent
import org.aresclient.ares.api.instruments.Module
import org.aresclient.ares.api.render.TextColor
import org.aresclient.ares.impl.util.ChatUtil

object MsgOnToggle: Module(Category.MISC, "MsgOnToggle", "Sends a chat message when a module is toggled") {
    @field:EventHandler
    private val toggleListener = EventListener<ToggleEvent> { event ->
        if(event.module == ClickGUI) return@EventListener
        ChatUtil.print("${if(event.enabled) "${TextColor.GREEN}Enabled" else "${TextColor.RED}Disabled"} ${TextColor.BLUE}${event.module.name}")
    }
}
